src/test/java/com/commercetools/sync/customers/utils/AddressUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1288 @@
+package com.commercetools.sync.customers.utils;
+
+import com.commercetools.sync.customers.commands.updateactions.AddBillingAddressIdWithKey;
+import com.commercetools.sync.customers.commands.updateactions.AddShippingAddressIdWithKey;
+import com.commercetools.sync.customers.commands.updateactions.SetDefaultBillingAddressWitKey;
+import com.commercetools.sync.customers.commands.updateactions.SetDefaultShippingAddressWitKey;
+import com.neovisionaries.i18n.CountryCode;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.customers.Customer;
+import io.sphere.sdk.customers.CustomerDraft;
+import io.sphere.sdk.customers.CustomerDraftBuilder;
+import io.sphere.sdk.customers.commands.updateactions.AddAddress;
+import io.sphere.sdk.customers.commands.updateactions.ChangeAddress;
+import io.sphere.sdk.customers.commands.updateactions.RemoveAddress;
+import io.sphere.sdk.customers.commands.updateactions.RemoveBillingAddressId;
+import io.sphere.sdk.customers.commands.updateactions.RemoveShippingAddressId;
+import io.sphere.sdk.models.Address;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildAddAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildAddBillingAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildAddShippingAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildAllAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildChangeAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildRemoveAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildRemoveBillingAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildRemoveShippingAddressUpdateActions;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildSetDefaultBillingAddressUpdateAction;
+import static com.commercetools.sync.customers.utils.CustomerUpdateActionUtils.buildSetDefaultShippingAddressUpdateAction;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AddressUpdateActionUtilsTest {
+
+    @Test
+    void buildAllAddressUpdateActions_WithDifferentAddresses_ShouldReturnAddressAction() {
+        final Customer oldCustomer = mock(Customer.class);
```

Comment:
Can declare oldCustomer as global variable. Then initialise it by `beforeEach` annotation

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/591/commits/f9e844c023a89327961f36479fb73a970223a732